docker/Dockerfile.nogil

```diff
@@ -0,0 +1,47 @@
+##############################################################
+# Builds a base image with free-threaded Python 3.13
+##############################################################
+
+ARG FROM_IMAGE=ubuntu:22.04
+
+# Builder stage: build Python 3.13 from source with the GIL disabled
+FROM $FROM_IMAGE AS builder
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Install build dependencies for Python
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    git \
+    libffi-dev \
+    libssl-dev \
+    zlib1g-dev
+
+# Clone CPython repository at the 3.13 branch
+RUN git clone --depth 1 --branch 3.13 https://github.com/python/cpython.git /usr/src/cpython
+
+WORKDIR /usr/src/cpython
+
+# Configure, build, and install Python 3.13 with the GIL disabled
+RUN ./configure --prefix=/opt/python3 --disable-gil --enable-optimizations
+RUN make -j$(nproc)
+RUN make install
+
+# Base runtime stage
```

Comment:
You don't need two stage in this one, just clone into `/tmp` then `rm -rf /tmp/*`. THis will avoid having to reinstall things like `libffi-dev`